src/lib/pmm.zig

```diff
@@ -76,10 +76,10 @@ pub fn switch_state(new_state: pmm_state) void {
   var current_base = sabaton.near("pmm_head").read(u64);
   current_base += page_size - 1;
   current_base &= ~(page_size - 1);
-
+  
   const eff_idx = @as(usize, @enumToInt(current_state)) * 3;
   const current_entry = sabaton.near("dram_base").addr(u64);
-
+  current_entry[eff_idx + 0] = dram;
```

Comment:
Are you sure about this? Doesn't this set every memmap entry to start at the dram base?